Sources/DeepLearning/Optimizer.swift

```diff
@@ -23,10 +23,8 @@ import TensorFlow
 public protocol Optimizer {
     /// The type of the model whose parameters are optimized.
     associatedtype Model: Layer
-    /// The scalar parameter type.
-    associatedtype Scalar: FloatingPoint
     /// The learning rate.
-    var learningRate: Scalar { get set }
+    var learningRate: Float { get set }
```

Comment:
I don't think we should change the optimizer protocol, since it's generic over not just tensor optimizers. Changing just the concrete non-Riemann optimizers would do what we need.

Reply:
Reverted.